applications/crossbar/src/modules/cb_braintree.erl

```diff
@@ -223,6 +229,18 @@ validate_credits(Context, ?HTTP_PUT) ->
                               maybe_charge_billing_id(Amount, Context1)
                       end.
 
+validate_token(Context, ?HTTP_GET) ->
+    try braintree_client_token:get_client_token(cb_context:account_id(Context)) of
+        Token ->
+            JObj = kz_json:set_value(<<"client_token">>, Token, kz_json:new()),
+            crossbar_util:response(JObj, Context)
+    catch
+        'throw':{'api_error', Reason} ->
+            crossbar_util:response('error', <<"braintree api error">>, 400, Reason, Context);
```

Comment:
Unless you differentiate errors, I would say a 5xx error is more appropriate (as the error happens on the server side). If you can add clauses that separate out errors that are due to client-provided information (like bad account id) as 4xx, and leave all others as 5xx (braintree API server down, for instance), that would be even better.